presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUpgrader.java

```diff
@@ -69,16 +74,57 @@ private static void upgradeUsingRpm(String pathToRpm, boolean checkDependencies)
             throws PrestoManagerException
     {
         checkRpmPackage(pathToRpm);
-        int upgradeRpm;
-        if (checkDependencies) {
-            upgradeRpm = executeCommand(SUBPROCESS_TIMEOUT, "sudo", "rpm", "-U", pathToRpm);
+        File configFiles = tarConfigFiles();
+        try {
+            int upgradeRpm;
+            if (checkDependencies) {
```

Comment:
This preserve config files should be a flag like --nodeps. Let user to choose as he/she may want a fresh copy to start over